pkg/bzz/address.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bzz
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/swarm"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var ErrInvalidAddress = errors.New("invalid address")
+
+type Address struct {
+	Underlay  ma.Multiaddr
+	Overlay   swarm.Address
+	Signature []byte
+}
+
+type addressJSON struct {
+	Overlay   string
+	Underlay  string
+	Signature string
+}
```

Comment:
JSON struct tags missing.

Reply:
Ah, I added it in a later commit, sorry :)